pkg/controllers/user/logging/clusterlogging.go

```diff
@@ -113,3 +127,43 @@ func (c *ClusterLoggingSyncer) createOrUpdateClusterConfigMap() error {
 	}
 	return utils.UpdateConfigMap(loggingconfig.ClusterConfigPath, loggingconfig.ClusterLoggingName, "cluster", c.configmaps)
 }
+
+func updateEmbeddedEndpoint(podLister v1.PodLister, serviceLister v1.ServiceLister, clusterLoggings v3.ClusterLoggingInterface) {
+	for {
+		select {
+		case <-timeout:
+			logrus.Warn("update embedded endpoint timeout")
+			return
+		case <-tick:
+			cls, err := clusterLoggings.List(metav1.ListOptions{})
+			if err != nil {
+				logrus.Debugf("get cluterlogging failed, %v", err)
```

Comment:
Do we proceed or return when the error occurs

Reply:
proceed. it will return when timeout or success set the endpoint